Add Inventory getters that do not initialize the inventory

nikitalita opened the pull request:
The implementation for `GetInventoryChanges()` in CLIB does not match that of the game; the game's does not attempt to initialize the inventory if it hasn't been already.

Initializing the inventory can have side-effects, like initializing the leveled items. This is undesirable in many circumstances. In order to avoid this, we have to add "NoInit" versions of the calls that end up calling `GetInventoryChanges()` (like `GetInventory()`, `GetGoldAmount()`, `GetWornArmor()`, etc.

The reason for not changing the current implementation of `GetInventoryChanges` is that I know that more than a few mods depend on `GetInventoryChanges()` initializing the inventory if it hasn't already, and I did not want to break those. If you want me to, I can just change this around so that the default calls don't initialize the inventory, or change signatures, etc.

powerof3 commented:
Not sure about these extra functions, feels like clutter. Perhaps `GetInventory(bool a_init = true)`?

nikitalita commented:
That’s fine; the main reason that I didn’t do that was that I wasn’t sure that if it would break ABI compatibility, but I don’t think that’s a concern here

nikitalita commented:
I went back and I changed all those functions to have a `no_init` parameter that defaults to `false` instead. I could not add `no_init` to certain overloaded functions like `GetInventory()` because it would make the current usage of `GetInventory(filter)` ambiguous; instead, I added a `DEFAULT_INVENTORY_FILTER` static constexpr to ease the use of `GetInventory(filter, no_init)`.

nikitalita committed:
Add `no_init` param to inventory getters

The implementation for GetInventoryChanges in CLIB does not match that of the game; the game's does not attempt to initialize the inventory if it hasn't been already.

Initializing the inventory can have side-effects, like initializing the leveled items. This is undesirable in many circumstances, so we add a `no_init` parameter to any function that gets the inventory to prevent this.

I've set `no_init` to false by default to prevent breaking any mod relying on the current behavior

nikitalita committed:
add static default inventory filter

This will make it easier to use the inventory getters without `no_init`
